### DISCUSSION TITLE
New use-case for https://containerssh.io/usecases/lab/ ( self-hosted cloudshell )

### DISCUSSION THREAD

**Discussion by kisst** [ContainerSSH/ContainerSSH#548] [Ideas]
As some of the bug report already shows  there is an other use-case which isn't listed and can enable a larger user-group to be involved in the project.
At Enterprise level having your own little box is still a need of the development and admin team,  hence random VM's do show up everywhere, but they are hard to get under control from both cost and admin point of view, hence the cloud vendor's cloudshell's are getting popular but their are far from perfect.
As far as I see the capabilities of ContainerSSH I think there is a real potential to be a solution for this use-case, of course some of the 0.5.0 version items like SSO and portfw would be crucial to get done before it can become use-full but before I start trialling, and  raising issues, and PR's I want to understand that if the project see this as a real use-case, or if I try to use it for this use-case it will be always twisting it's arm.

The expectation for self hosted cloudshell would be something like this:
 - Self hosted ( this is crucial as most of the cases you need special internal network access ) 
 - On demand ( the container run only when there is a user logged in )
 - Can run on custom images ( so the ops team can prep images which have all pre-installed tools a user might need ) 
 - Have persistent storage ( user's want to have their own code / script stored, so NFS or similar would be needed ) 
 
 As far as I see, containerssh can provide most of this, yet I don't see it listed on the use-cases... Hence asking.

**Comment by ghost** [comment]
Hey @kisst the project was indeed used in this fashion in the CERN pilot project: https://www.youtube.com/watch?v=d6aCBiInMbg
